src/android/FirebasePlugin.java

```diff
@@ -724,13 +724,17 @@ public void onVerificationCompleted(PhoneAuthCredential credential) {
                             // 2 - Auto-retrieval. On some devices Google Play services can automatically
                             //     detect the incoming verification SMS and perform verificaiton without
                             //     user action.
-                            Log.d(TAG, "success: verifyPhoneNumber.onVerificationCompleted - callback and create a custom JWT Token on server and sign in with custom token - we cant do anything");
+                            Log.d(TAG, "success: verifyPhoneNumber.onVerificationCompleted");
 
                             JSONObject returnResults = new JSONObject();
                             try {
-                                returnResults.put("verificationId", false);
+                                String verificationId = getPrivateField(credential, "zzfc");
+                                String code = getPrivateField(credential, "zzfd");
+
+                                returnResults.put("verificationId", verificationId);
+                                returnResults.put("code", code);
                                 returnResults.put("instantVerification", true);
-                            } catch (JSONException e) {
+                            } catch(JSONException | IllegalAccessException | NoSuchFieldException e){
```

Comment:
I would wrap each private field invocation with it's own try catch as it is unsafe to rely on obfuscated property names.  These can change without warning.  If they did change, ALL callbacks would not work. If you wrap them each in try catch, then at least you still get the public information if the private fields changed.

Reply:
You are right about the obfuscated properties. I'm not quite sure I understand your concern about the callbacks though, the error message gets passed back to cordova with the callbackContext.error if I'm not mistaken. Results would be similar if I wrapped them each since the first exception would lead to error being called.

Reply:
Lets say that Firebase decides to rename 'zzfd'.  If this happens, the current code will throw an error.  I don't believe that should be the behavior.  Instead, I would say that we include any private fields if they exist, but don't throw an error if one of the private fields does not exist.  We can do this by catching the error in a try/catch when accessing the private fields.  If an exception is thrown, we ignore that field and move onto the next field.  This way we don't fail the entire call if one private field is missing.

Not everyone is looking for the private fields and there is no reason their code should fail if a private field changes.  Does that make sense?

Reply:
I improved it a bit to iterate through all properties to still be compatible in case of name changes. I've noticed that there are only 4 properties of type String. Two of them are null. VerificationId seems to be constant at 140 chars, code at 6. So we can easily identify which is which.
Also changed the behavior a bit. If either one can't be accessed we return false. Let me know what you think.